effekt/jvm/src/test/scala/effekt/core/InterpreterTests.scala

```diff
@@ -0,0 +1,304 @@
+      |  def inc(): Unit
+      |  def get(): Int
+      |}
+      |
+      |def main() = {
+      |  def c = new Counter {
+      |    def inc() = println("tick")
+      |    def get() = 0
+      |  };
+      |  c.inc()
+      |  c.inc()
+      |  c.inc()
+      |}
+      |
+      |""".stripMargin
+
+  val factorialAccumulator =
+    """import examples/benchmarks/runner
+      |
+      |def factorial(a: Int, i: Int): Int =
+      |  if (i == 0) {
+      |    a
+      |  } else {
+      |    factorial((i * a).mod(1000000007), i - 1)
+      |  }
+      |
+      |def run(n: Int): Int =
+      |  factorial(1, n)
+      |
+      |def main() = benchmark(5){run}
+      |
+      |""".stripMargin
+
+  val sort =
+    """import list
+      |
+      |def main() = {
+      |  // synchronized with doctest in `sortBy`
+      |  println([1, 3, -1, 5].sortBy { (a, b) => a <= b })
+      |}
+      |""".stripMargin
+
+  val mutableState =
+    """def main() = {
+      |  var x = 42;
+      |  x = x + 1;
+      |  println(x.show)
+      |  [1, 2, 3].map { x => x + 1 }.foreach { x => println(x) };
+      |
+      |  region r {
+      |    var x in r = 42;
+      |    x = x + 1
+      |    println(x)
+      |  }
+      |}
+      |""".stripMargin
+
+  val simpleException =
+    """effect raise(): Unit
+      |
+      |def main() = {
+      |  try {
+      |    println("before");
+      |    do raise()
+      |    println("after")
+      |  } with raise { println("caught") }
+      |}
+      |
+      |""".stripMargin
+
+  val triples =
+    """import examples/benchmarks/runner
+      |
+      |record Triple(a: Int, b: Int, c: Int)
+      |
+      |interface Flip {
+      |  def flip(): Bool
+      |}
+      |
+      |interface Fail {
+      |  def fail(): Nothing
+      |}
+      |
+      |def choice(n: Int): Int / {Flip, Fail} = {
+      |  if (n < 1) {
+      |    do fail() match {}
+      |  } else if (do flip()) {
+      |    n
+      |  } else {
+      |    choice(n - 1)
+      |  }
+      |}
+      |
+      |def triple(n: Int, s: Int): Triple / {Flip, Fail} = {
+      |  val i = choice(n)
+      |  val j = choice(i - 1)
+      |  val k = choice(j - 1)
+      |  if (i + j + k == s) {
+      |    Triple(i, j, k)
+      |  } else {
+      |    do fail() match {}
+      |  }
+      |}
+      |
+      |def hash(triple: Triple): Int = triple match {
+      |  case Triple(a, b, c) => mod(((53 * a) + 2809 * b + 148877 * c), 1000000007)
+      |}
+      |
+      |def run(n: Int) =
+      |  try {
+      |    hash(triple(n, n))
+      |  } with Flip {
+      |    def flip() = mod(resume(true) + resume(false), 1000000007)
+      |  } with Fail {
+      |    def fail() = 0
+      |  }
+      |
+      |def main() = benchmark(10){run}
+      |
+      |
+      |""".stripMargin
+
+  // doesn't work: product_early (since it SO due to run run run)
+
+  val Some(main, mod, decl) = runFile("examples/benchmarks/effect_handlers_bench/triples.effekt"): @unchecked
```

Comment:
You can already experiment with it by changing the filename here and then running `testOnly effekt.core.InterpreterTests`

Reply:
For example, here is the result of running `triples`:

```
Static dispatches: 232
Dynamic dispatches: 231
Branches: 701
Pattern matches: 5
Frames (pushed: 405, popped: 530)
Allocations: 5
Closures: 112
Field lookups: 0
Variable reads: 0
Variable writes: 0
Installed delimiters: 1
Captured continuations: 347
Resumed continuations: 350
```